src/micromerge.ts

```diff
@@ -1180,6 +1032,132 @@ export default class Micromerge {
+            // Once we are DURING the operation, we'll start a patch, emitting an intermediate patch
+            // any time the formatting changes during that range, and eventually emitting one last patch
+            // at the end of the range (or document.) 
+            if (side === "markOpsAfter" && !elMeta.deleted) {
+                // We need to keep track of the "visible" index, since the outside world won't know about
+                // deleted characters.
+                visibleIndex += 1
+            }
+
+            if (changedOps) {
+                // First see if we need to emit a new patch, which occurs when formatting changes
+                // within the range of characters the formatting operation targets.
+                if (partialPatch) {
+                    const patch = this.finishPartialPatch(partialPatch, visibleIndex, objLength)
+                    if (patch) { patches.push(patch) }
+                    partialPatch = undefined
+                }
+
+                // Now begin a new patch since we have new formatting to send out.
+                if (opState == "DURING" && !isEqual(opsToMarks(currentOps), opsToMarks(changedOps))) {
+                    partialPatch = this.beginPartialPatch(op, visibleIndex)
+                }
+            }
+
+            if (opState == "AFTER") { break }
+        }
+
+        // If we have a partial patch leftover at the end, emit it
+        if (partialPatch) {
+            const patch = this.finishPartialPatch(partialPatch, visibleIndex, objLength)
+            if (patch) { patches.push(patch) }
+        }
+
+        return patches
+    }
+
+    private calculateOpsForPosition(
+        op: MarkOperation, currentOps: Set<MarkOperation>,
+        side: "markOpsBefore" | "markOpsAfter",
+        elMeta: ListItemMetadata,
+        opState: MarkOpState): [opState: MarkOpState, newOps?: Set<MarkOperation>] {
+        // Compute an index in the visible characters which will be used for patches.
+        // If this character is visible and we're on the "after slot", then the relevant
+        // index is one to the right of the current visible index.
+        // Otherwise, just use the current visible index.
+        const opSide = side === "markOpsAfter" ? "after" : "before"
+
+        if (op.start.type === opSide && op.start.elemId === elMeta.elemId) {
+            // we've reached the start of the operation
+            return ["DURING", new Set([...currentOps, op])]
+        } else if (op.end.type === opSide && op.end.elemId === elMeta.elemId) {
+            // and here's the end of the operation
+            return ["AFTER", new Set([...currentOps].filter(opInSet => opInSet !== op))]
+        } else if (opState == "DURING" && elMeta[side] !== undefined) {
+            // we've hit some kind of change in formatting mid-operation
+            return ["DURING", new Set([...currentOps, op])]
+        }
+
+        // No change...
+        return [opState, undefined]
```

Comment:
@pvh what would you think of returning `currentOps` in the case of no change, so that we always just return the updated ops for this position?

Reply:
Ah, I guess that doesn't work great for emitting patches, so maybe better to leave like this